src/components/LmsConfigurations/index.test.jsx

```diff
@@ -39,37 +47,85 @@ REQUIRED_CANVAS_CONFIG_FIELDS.forEach((field) => {
 });
 
+REQUIRED_BLACKBOARD_CONFIG_FIELDS.forEach((field) => {
+  formData.append(field, 'testdata');
+  blackboardResponse.data.results[0][field] = 'testdata';
+});
+
 const waitForAsync = () => new Promise(resolve => setImmediate(resolve));
 
 describe('<LmsConfigurations /> ', () => {
-  it('get LMS configurations when present', async () => {
+  it('get the Moodle LMS configuration when present', async () => {
     LmsApiService.fetchMoodleConfig.mockResolvedValue(moodleResponse);
-    LmsApiService.fetchCanvasConfig.mockResolvedValue(canvasResponse);
+
+    // Mock the fetch of the other configs to 404, not found
+    LmsApiService.fetchBlackboardConfig.mockRejectedValue(notFoundBlackboardResponse);
+    LmsApiService.fetchCanvasConfig.mockRejectedValue(notFoundCanvasResponse);
+
     const wrapper = mount(<LmsConfigurations enterpriseId="testEnterpriseId" />);
     await waitForAsync();
     wrapper.update();
     expect(wrapper.state().moodleConfig).toEqual(moodleResponse.data.results[0]);
     expect(wrapper.find('MoodleIntegrationConfigForm').props().config).toEqual(wrapper.state().moodleConfig);
   });
 
-  it('return Error Page when any LMS fetch request fails with error (not 404)', async () => {
+  it('get the Blackboard LMS configuration when present', async () => {
+    LmsApiService.fetchBlackboardConfig.mockResolvedValue(blackboardResponse);
+
+    // Mock the fetch of the other configs to 404, not found
+    LmsApiService.fetchMoodleConfig.mockRejectedValue(notFoundMoodleResponse);
+    LmsApiService.fetchCanvasConfig.mockRejectedValue(notFoundCanvasResponse);
+
+    const wrapper = mount(<LmsConfigurations enterpriseId="testEnterpriseId" />);
+    await waitForAsync();
+    wrapper.update();
+    expect(wrapper.state().blackboardConfig).toEqual(blackboardResponse.data.results[0]);
+    expect(wrapper.find('BlackboardIntegrationConfigForm').props().config).toEqual(wrapper.state().blackboardConfig);
+  });
+
+  it('return Error Page when the Moodle LMS fetch request fails with error (not 404)', async () => {
     LmsApiService.fetchMoodleConfig.mockRejectedValue(badMoodleResponse);
-    LmsApiService.fetchCanvasConfig.mockRejectedValue(badCanvasResponse);
+
+    // We want to ensure that the Moodle fetch config method will cause the error page to display
+    // if it encounters an unexpected error while the other config fetch methods succeed.
+    LmsApiService.fetchBlackboardConfig.mockResolvedValue(blackboardResponse);
+    LmsApiService.fetchCanvasConfig.mockResolvedValue(canvasResponse);
+
     const wrapper = mount(<LmsConfigurations enterpriseId="testEnterpriseId" />);
     await waitForAsync();
     wrapper.update();
+
+    // Expect the moodle config to be false, while the others to be true
     expect(wrapper.state().moodleConfig).toBeFalsy();
     expect(wrapper.find('ErrorPage').length).toBe(1);
   });
 
+  it('return Error Page when Blackboard LMS fetch request fails with an error (not 404)', async () => {
```

Comment:
oh I think this was my b, but I really don't see a reason to have individual tests for 1 missing config. The result is the same regardless of which config is missing (an error page is shown), so it seems a bit redundant.

When a config is present however, the actual result of that should be different for each LMS, which is why individual tests to verify the Canvas/Moodle/Blackboard/etc. component is rendered with the config are valuable.

But this? We can just do one test together to verify it shows an error when any config fails with a 400/>404
However, if it's a pain I can just clean up in my PR. I really don't care.

Reply:
Right, but we want to verify that the response actually *is* the same if any config is missing. Also, if we decide to refactor and change how we do things, testing each one will ensure that we don't do something that lets one of them slide. Having individual tests lets us easily identify where the problem is. It is a bit redundant, but personally I like redundancy in tests.

Reply:
I really don't like redundancy in tests mostly because of it slowing down feedback from CI and I feel like the way we created these APIs there's not really a way that the results would vary at all between the API endpoints, but regardless, I'd rather not keep having PRs blocked while arguing about it and our build times for this repo are quite fast at the moment, so we should just do it @alex-sheehan-edx.